src/java.net.http/share/classes/jdk/internal/net/http/Http3Connection.java

```diff
@@ -126,11 +126,16 @@ public final class Http3Connection implements AutoCloseable {
     // as per spec
     // -1 is used to imply no GOAWAY received so far
     private final AtomicLong lowestGoAwayReceipt = new AtomicLong(-1);
+
+    private final Duration idleTimeoutDuration;
     private volatile IdleConnectionTimeoutEvent idleConnectionTimeoutEvent;
     // value of true implies no more streams will be initiated on this connection,
     // and the connection will be closed once the in-progress streams complete.
     private volatile boolean finalStream;
     private volatile boolean allowOnlyOneStream;
+    // true if this connection has been placed in the HTTP/3 connection pool of the HttpClient.
+    // false otherwise.
+    private volatile boolean presentInConnPool;
```

Comment:
In theory a connection that is not in the pool should be marked as `finalStream` ;